server/server.go

```diff
@@ -3432,6 +3440,47 @@ func (s *Server) createClientEx(conn net.Conn, inProcess bool) *client {
 		}
 	}
 
+	// Check for proxy protocol if enabled.
+	if !isClosed && !tlsRequired && opts.ProxyProtocol {
+		if len(pre) == 0 {
+			// There has been no pre-read yet, do so so we can work out
+			// if the client is trying to negotiate PROXY.
+			pre = make([]byte, 6)
+			c.nc.SetReadDeadline(time.Now().Add(proxyProtoReadTimeout))
```

Comment:
Looks like readProxyProtoHeader() invoked below is going to do a read. Granted, it would be from tlsMixConn that will already have the pre buffer read, so there won't be an actual io read, but I wonder if this is necessary to do it here.